components/docker/component.go

```diff
@@ -141,7 +141,8 @@ func (d *Manager) ComposeStrUp(name string, composeManifests []ComposeInlineMani
 }
 
 func (d *Manager) install() (*remote.Command, error) {
-	dockerInstall, err := d.Host.OS.PackageManager().Ensure("docker.io", nil, "docker", d.opts...)
+	opts := append(d.opts, pulumi.Parent(d))
```

Comment:
This may modify `d.opts` depending on whether there is available capacity in `d.opts` or not, that's why it's important to always use `utils.MergeOptions`

Reply:
TIL: `append` modifies the passed slice even if we assign the result to a new slice